### ISSUE TITLE
修复klein多图参考训练报错

### ISSUE THREAD

**Issue by JianfengZhang112358** [lrzjason/T2ITrainer#98]
从2138行开始修改
原来的代码：
try:
    cond_model_input = torch.stack(reference_list, dim=0) if len(reference_list) > 0 else None
except:
    # print image file path
    # rollback to single image reference
    cond_model_input = torch.stack([reference_list[0]], dim=0) if len(reference_list) > 0 else None
    for i in range(len(reference_list)):
        print("reference_list cat error size:", reference_list[i].shape)
    print("original image path ", batch_targets[0]['original_image_path'])
if len(reference_list) > 0:
    # cond_model_input_list = [cond_model_input[i].unsqueeze(0) for i in range(cond_model_input.shape[0])]
    ref_latents_ids = Flux2KleinPipeline._prepare_image_ids(reference_list).to(
        device=accelerator.device
    )
    ref_latents_ids = ref_latents_ids.view(
        ref_latents_ids.shape[0], -1, model_input_ids.shape[-1]
    )
    packed_ref_latents = Flux2KleinPipeline._pack_latents(cond_model_input)
    # convert packed_ref_latents from (x, d, c) to (1, x*d, c)
    packed_ref_latents = packed_ref_latents.reshape(model_input.shape[0], -1, packed_ref_latents.shape[-1])
    
    # print("model_input.shape", model_input.shape)
    # print("packed_ref_latents.shape", packed_ref_latents.shape)
    
    model_input = torch.cat([model_input, packed_ref_latents], dim=1)
    model_input_ids = torch.cat([model_input_ids, ref_latents_ids], dim=1)
修改后代码：
try:
    # cond_model_input = torch.stack(reference_list, dim=0) if len(reference_list) > 0 else None
    #当多张参考图大小不一样时，需要先对每张cond latents进行_pack_latents，再在token维度拼接
    packed_ref_latents_list = []
    for cond_latents in reference_list:
        cond_latents = cond_latents.unsqueeze(0)#前面默认bs=1把bs维度压缩了
        packed_cond_latents = Flux2KleinPipeline._pack_latents(cond_latents)
        packed_ref_latents_list.append(packed_cond_latents)
    packed_ref_latents = torch.cat(packed_ref_latents_list, dim=1) #在token维度拼接参考图的latents
except:
    # print image file path
    # rollback to single image reference
    cond_model_input = torch.stack([reference_list[0]], dim=0) if len(reference_list) > 0 else None
    for i in range(len(reference_list)):
        print("reference_list cat error size:", reference_list[i].shape)
    print("original image path ", batch_targets[0]['original_image_path'])
if len(reference_list) > 0:
    # cond_model_input_list = [cond_model_input[i].unsqueeze(0) for i in range(cond_model_input.shape[0])]
    ref_latents_ids = Flux2KleinPipeline._prepare_image_ids(reference_list).to(
        device=accelerator.device
    )
    ref_latents_ids = ref_latents_ids.view(
        ref_latents_ids.shape[0], -1, model_input_ids.shape[-1]
    )
    # packed_ref_latents = Flux2KleinPipeline._pack_latents(cond_model_input)
    # # convert packed_ref_latents from (x, d, c) to (1, x*d, c)
    # packed_ref_latents = packed_ref_latents.reshape(model_input.shape[0], -1, packed_ref_latents.shape[-1])
    
    # print("model_input.shape", model_input.shape)
    # print("packed_ref_latents.shape", packed_ref_latents.shape)
    
    model_input = torch.cat([model_input, packed_ref_latents], dim=1)
    model_input_ids = torch.cat([model_input_ids, ref_latents_ids], dim=1)

**Comment by Z0Victor** [comment]
bro, how to train klein LoRA, Do you have lecture? what script I should use? THANKS!
